GliaWidgetsTests/Sources/ChatViewModelTests.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ChatViewModelTests.swift
+//  GliaWidgetsTests
+//
+//  Created by Yurii Dukhovnyi on 18.02.2022.
+//
+
+import XCTest
+
+@testable import GliaWidgets
+
+class ChatViewModelTests: XCTestCase {
+
+    var viewModel: ChatViewModel!
+
+    func test__choiceOptionSelected() throws {
+
+        enum Call { case sendSelectedOptionValue }
+        var calls = [Call]()
+
+        var fileManager = FoundationBased.FileManager.failing
+        fileManager.urlsForDirectoryInDomainMask = { _, _ in [URL(fileURLWithPath: "/i/m/mocked/url")] }
+        fileManager.createDirectoryAtUrlWithIntermediateDirectories = { _, _, _ in }
+
+        viewModel = .init(
+            interactor: .mock(),
+            alertConfiguration: .mock(),
+            screenShareHandler: ScreenShareHandler(),
+            call: .init(with: nil),
+            unreadMessages: .init(with: 0),
+            showsCallBubble: true,
+            isWindowVisible: .init(with: true),
+            startAction: .none,
+            environment: .init(
+                chatStorage: .failing,
+                fetchFile: { _, _, _ in },
+                sendSelectedOptionValue: { _, _ in
+                    calls.append(.sendSelectedOptionValue)
+                },
+                uploadFileToEngagement: { _, _, _ in },
+                fileManager: fileManager,
+                data: .failing,
+                date: { Date() }
```

Comment:
I would like to suggest to use mock, like Date.mock or something similar if possible. This is because date is a side-effect, and if for example we are to have some snapshot-based tests, it can cause problems.

Reply:
fixed - 26afd5001506dff4d7261fb2783e8d0ba19df698